client/src/components/App/Content/SignUp/SignUpForm/SignUpForm.tsx

```diff
@@ -0,0 +1,348 @@
+import Language from 'models/Language'
+import AlphabetTextField from 'commons/AlphabetTextField/AlphabetTextField';
+import NumericTextField from 'commons/NumericTextField/NumericTextField'
+import FormInput from 'commons/FormInput/FormInput'
+import StoreStateType from 'redux/storeStateType';
+import { get } from 'Utils/auxiliaryFunctions/auxiliaryFunctions'
+
+import SignUpSchema from './SignUpSchema'
+import useSignUpForm from './useSignUpForm'
+import useStyles from './SignUpFormStyles'
+
+
+const MABAR_USER_NAME = 'שם משתמש מב"ר'
+const FIRST_NAME_LABEL = 'שם פרטי'
+const LAST_NAME_LABEL = 'שם משפחה'
+const CITY_LABEL = 'עיר'
+const PHONE_NUMBER_LABEL = 'מספר טלפון'
+const ID_LABEL = 'תז'
+const MAIL_LABEL = 'מייל'
+const COUNTY_LABEL = 'נפה'
+const SOURCE_ORGANIZATION_LABEL = 'מסגרת'
+const LANGUAGE_LABEL = 'שפה'
+
+const SignUpForm: React.FC<Props> = ({ handleSaveUser }: Props) => {
+    const classes = useStyles();
+    
+    const [counties, setCounties] = useState<County[]>([]);
+    const [languages, setLanguages] = useState<Language[]>([]);
+    const [sourcesOrganization, setSourcesOrganization] = useState<SourceOrganization[]>([])
+    
+    const { getDefaultValues, createUser } = useSignUpForm({ setCounties, setSourcesOrganization, setLanguages,
+                                                             handleSaveUser });
+
+    const cities = useSelector<StoreStateType, Map<string, City>>(state => state.cities);
+    
+    const methods = useForm<SignUpUser>({
+        mode: 'all',
+        defaultValues: getDefaultValues(),
+        resolver: yupResolver(SignUpSchema)
+    })
+
+    const onSubmit = (data: SignUpUser) => {
+        createUser(data);
+    }
+
+    return (
+        <FormProvider {...methods}>
+            <form id='signUp' onSubmit={methods.handleSubmit(onSubmit)}>
+                <Grid container justify='flex-start' className={classes.formRow}>
+                    <Grid item xs={8}>
+                        <FormInput fieldName='שם משתמש מב"ר'>
+                            <Controller 
+                                name={SignUpFields.MABAR_USER_NAME}
+                                control={methods.control}
+                                render={(props) => (
+                                    <TextField
+                                        test-id={props.name}
+                                        value={props.value}
+                                        onChange={(event) => props.onChange(event.target.value as string)}
+                                        placeholder='itay.benmoshe'
+                                        onBlur={props.onBlur}
+                                        error={get(methods.errors, props.name)}
+                                        label={get(methods.errors, props.name)? 
+                                                get(methods.errors, props.name).message : 
+                                                MABAR_USER_NAME}
```

Comment:
```suggestion
                                        label={
                                        get(methods.errors, props.name)?.message || 									
                                        MABAR_USER_NAME
                                        }
```

could make condition shorter instead of calling get() method twice

Reply:
relevant to all labels

Reply:
fixed